helm/cluster/templates/clusterapi/_helpers_files.tpl

```diff
@@ -124,13 +124,29 @@ and is used to join the node to the teleport cluster.
 {{/* Provider-specific files for all nodes */}}
 {{- define "cluster.internal.kubeadm.files.provider" }}
 {{- if $.Values.providerIntegration.kubeadmConfig.files }}
-{{ toYaml $.Values.providerIntegration.kubeadmConfig.files }}
+{{ include "cluster.processFiles" (dict "files" $.Values.providerIntegration.kubeadmConfig.files "clusterName" (include "cluster.resource.name" $)) }}
 {{- end }}
 {{- end }}
 
 {{/* Custom cluster-specific files for all nodes */}}
 {{- define "cluster.internal.kubeadm.files.custom" }}
 {{- if $.Values.internal.advancedConfiguration.files }}
-{{ toYaml $.Values.internal.advancedConfiguration.files }}
+{{ include "cluster.processFiles" (dict "files" $.Values.internal.advancedConfiguration.files "clusterName" (include "cluster.resource.name" $)) }}
 {{- end }}
 {{- end }}
+
+{{- define "cluster.processFiles" }}
```

Comment:
Just an idea but maybe it would make sense to align with the other define functions here and name it `cluster.internal.processFiles`? Thoughts @nprokopic ?